scenes/globals/pause.gd

```diff
@@ -0,0 +1,88 @@
+extends Node
+
+enum System {
+	PLAYER_INPUT,
+	GAME,
+}
+
+var _pause_requests: Dictionary[System, Array] = {}
+
+# This should be an array of systems but it throws
+# a runtime error when using the filter function, so for now just "Array"
+var _paused_systems: Array = []
```

Comment:
The `_paused_systems` is a variable derived from `_pause_requests`, it is an Array built from the keys of the Dictionary. This is prone to errors and has to be constantly sync'ed. I propose that we get rid of it and use the Dictionary keys directly.